lib/puppet/provider/dns_record/dynect.rb

```diff
@@ -0,0 +1,149 @@
+
+# Author:: Charles Dunbar <charles@puppetlabs.com>
+# Type Name:: dns_record
+# Provider:: dynect
+#
+# Copyright 2015, Puppet Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'json'
```

Comment:
Having requires outside the provider block can cause some issues. For example, if the dependencies aren't installed yet the run might fail because it will try to load them before the package resources are evaluated. Some people manage gem dependencies like this http://alcy.github.io/2012/11/21/handling-gem-dependencies-in-custom-puppet-providers/ where the gem is a feature of the provider. I didn't use this method in the chronos module because I think it makes more sense to manage them as package resources so they actually show up in the catalog.